.evergreen/venv-utils.sh

```diff
@@ -21,38 +21,73 @@
 #   "$2": The path to the virtual environment (directory) to create.
 #
 # Return 0 (true) if the virtual environment has been successfully created,
-# activated, and the pip package upgraded.
-# Return a non-zero value (false) otherwise.
+# activated, and all seed packages are successfully installed in the new
+# virtual environment.
+# Return a non-zero value (false) in a deactivated state otherwise.
+#
+# The "seed" packages pip, setuptools, and wheel are automatically installed
+# into the virtual environment. All packages must be successfully installed for
+# venvcreate to be considered a success.
 #
 # If a file or directory exists at the given path to the virtual environment,
 # they may be deleted as part of virtual environment creation.
 venvcreate() {
   local -r bin="${1:?'venvcreate requires a Python binary to use for the virtual environment'}"
   local -r path="${2:?'venvcreate requires a path to the virtual environment to create'}"
 
+  if [[ "$OSTYPE" == cygwin ]]; then
+    local -r real_path="$(cygpath -aw "$path")" || return
+  else
+    local -r real_path="$path" || return
+  fi
+
   # Prefer venv, but fallback to virtualenv if venv fails.
   for mod in "venv" "virtualenv"; do
-    # Ensure a clean directory before attempting to create a virtual environment.
+    # Ensure a clean directory before attempting to create a virtual
+    # environment.
     rm -rf "$path"
 
-    if "$bin" -m "$mod" "$path"; then
-      # Workaround https://bugs.python.org/issue32451:
-      # mongovenv/Scripts/activate: line 3: $'\r': command not found
-      if [[ -f "$path/Scripts/activate" ]]; then
-        dos2unix "$path/Scripts/activate" || return
-      fi
+    case "$mod" in
+    venv)
+      "$bin" -m "$mod" --system-site-packages "$real_path" || continue
+      ;;
+    virtualenv)
+      # -p: ensure correct Python binary is used by virtual environment.
```

Comment:
Is -p actually needed here? `-p` defaults to the current version of python so this seems redundant.

Reply:
Oh, can you add the comment from the old script? It's much more informative.

Reply:
Will do. 👍